python/hsfs/feature_group.py

```diff
@@ -1629,30 +1629,29 @@ def save(
             TypeVar("pyspark.sql.DataFrame"),  # noqa: F821
             TypeVar("pyspark.RDD"),  # noqa: F821
             np.ndarray,
-            List[list],
-        ],
+            List[feature.Feature],
+        ] = None,
         write_options: Optional[Dict[Any, Any]] = {},
         validation_options: Optional[Dict[Any, Any]] = {},
         wait: bool = False,
     ):
         """Persist the metadata and materialize the feature group to the feature store.
 
-        !!! warning "Deprecated"
-            `save` method is deprecated. Use the `insert` method instead.
-
         !!! warning "Changed in 3.3.0"
             `insert` and `save` methods are now async by default in non-spark clients.
             To achieve the old behaviour, set `wait` argument to `True`.
 
-        Calling `save` creates the metadata for the feature group in the feature store
-        and writes the specified `features` dataframe as feature group to the
+        Calling `save` creates the metadata for the feature group in the feature store.
+        If a DataFrame, RDD or Ndarray is provided, the data is written to the
         online/offline feature store as specified.
         By default, this writes the feature group to the offline storage, and if
         `online_enabled` for the feature group, also to the online feature store.
         The `features` dataframe can be a Spark DataFrame or RDD, a Pandas DataFrame,
         or a two-dimensional Numpy array or a two-dimensional Python nested list.
         # Arguments
-            features: Query, DataFrame, RDD, Ndarray, list. Features to be saved.
+            features: DataFrame, RDD, Ndarray or a list of features. Features to be saved.
```

Comment:
It might be nice to add the optional features list argument to `ExternalFeatureGroup.save()` as well.

Reply:
I'm not sure it makes sense to be honest. You'll always need to provide a query (or some other configuration options to read from S3 for instance) - i.e. you can't create an empty external feature group and then add the query later. 

Also you can already provide the set of features as part of the `create_external_feature_group` method.